src/tcgwars/logic/impl/gen7/SunMoonPromos.groovy

```diff
@@ -2859,20 +3010,162 @@ public enum SunMoonPromos implements LogicCardInfo {
         return copy(CosmicEclipse.PHIONE_57, this);
       case BLACEPHALON_SM221:
         return copy(CosmicEclipse.BLACEPHALON_104, this);
+      case MISMAGIUS_SM222:
+      return evolution (this, from:"Misdreavus", hp:HP110, type:P, retreatCost:1) {
+        weakness D
+        resistance F, MINUS20
+        move "Psywave", {
+          text "20x damage. This attack does 20 damage times the amount of Energy attached to your opponent's Active Pokémon."
+          energyCost P
+          onAttack {
+            damage 20*opp.active.cards.energyCount(C)
+          }
+        }
+        move "Astonish", {
+          text "40 damage. Choose a random card from your opponent's hand. Your opponent reveals that card and shuffles it into their deck."
+          energyCost C, C
+          onAttack {
+            damage 40
+            astonish()
+          }
+        }
+      };
       case TERRAKION_SM223:
         return copy(UnifiedMinds.TERRAKION_122, this);
+      case CELEBI_SM224:
+      return basic (this, hp:HP070, type:P, retreatCost:1) {
+        weakness P
+        move "Call for Greatness", {
+          text "Search your deck for up to 3 Pokémon-GX with different names, reveal them, and put them into your hand. Then, shuffle your deck."
+          energyCost C
+          attackRequirement {
+            assert my.deck : "Your deck is empty."
+          }
+          onAttack {
+            // TODO
+          }
+        }
+        move "Psy Bolt", {
+          text "40 damage. Flip a coin. If heads, your opponent’s Active Pokémon is now Paralyzed."
+          energyCost P, C, C
+          onAttack {
+            damage 40
+            flip { apply PARALYZED }
+          }
+        }
+      };
       case VICTINI_SM225:
         return copy(UnifiedMinds.VICTINI_26, this);
       case CHARIZARD_SM226:
         return copy(TeamUp.CHARIZARD_14, this);
+      case PIKACHU_SM227:
+        return basic (this, hp:HP060, type:L, retreatCost:1) {
+          weakness F
+          resistance M, MINUS20
+          move "Quick Attack", {
+            text "10+ damage. Flip a coin. If heads, this attack does 10 more damage."
+            energyCost C
+            onAttack {
+              damage 10
+              flip { damage 10 }
+            }
+          }
+          move "Thunderbolt", {
+            text "80 damage. Discard all Energy from this Pokémon."
+            energyCost L, L, C
+            onAttack {
+              damage 80
+              discardAllSelfEnergy(null)
+              }
```

Comment:
This appears to be an extra bracket

Reply:
This is the ending bracket for the `onAttack` field for Thunderbolt.